tests/unit_tests/test_heron.py

```diff
@@ -1,61 +1,669 @@
+    mock_open_mult = mock_open()
+    mock_open_mult.side_effect = [mock_open(read_data =
+                                  """{
+                                          "Component Set Name": "Component0",
+                                          "Reference Driver": 1000,
+                                          "Reference Driver Power Units": "mW",
+                                          "Reference Price (USD)": 1000,
+                                          "Scaling Factor": 0.1
+                                      }""").return_value,
+                                  mock_open(read_data =
+                                  """{
+                                          "Component Set Name": "Component1",
+                                          "Reference Driver": 2100,
+                                          "Reference Driver Power Units": "mW",
+                                          "Reference Price (USD)": 2200,
+                                          "Scaling Factor": 0.2
+                                      }""").return_value,
+                                  mock_open(read_data =
+                                  """{
+                                          "Component Set Name": "Component2",
+                                          "Reference Driver": 3100,
+                                          "Reference Driver Power Units": "mW",
+                                          "Reference Price (USD)": 3200,
+                                          "Scaling Factor": 0.3
+                                      }""").return_value,
+                                  mock_open(read_data =
+                                  """{
+                                          "Component Set Name": "Component3",
+                                          "Reference Driver": 4100,
+                                          "Reference Driver Power Units": "mW",
+                                          "Reference Price (USD)": 4200,
+                                          "Scaling Factor": 0.4
+                                      }""").return_value]
+                                    
+
+    # Call the function with patch for open function
+    with patch('builtins.open', mock_open_mult):
+      result_tree = create_componentsets_in_HERON("/fake/folder", "/fake/heron_input.xml")
+
+    # Verify Component nodes were merged
+    component_nodes = result_tree.findall("./Components/Component")
+    self.assertEqual(len(component_nodes), 4)
+
+    for comp in component_nodes:
+      if comp.get('name') == 'Component0':
+        economics = comp.find('./economics')
+        
+        # Verify non-capex cashflow was not corrupted
+        with self.subTest("Non-capex CashFlow node was corrupted"):
+          cashflow_non_capex = economics.findall('./CashFlow[@name="other"]')
+          self.assertEqual(len(cashflow_non_capex), 1)
+          cf_noncap_contents = [e for e in cashflow_non_capex[0].findall('./')
```

Comment:
Yes, the purpose of this (lines 277-278) is to filter the children of the `CashFlow[@name="other"]` node to find all non-comment subelements (i.e., subnodes). This filtering is necessary because the length of the resulting list is checked. A `getchildren()` method would be cleaner than `findall('./')`, but there is not one included in the `ET` package. The `iter()` or `iterfind()` methods could be used in place of the `findall()` method, but neither would simplify syntax; xml pathing (the `'./'` argument) would still be required. Documentation here: https://docs.python.org/3/library/xml.etree.elementtree.html#xml.etree.ElementTree.Element

Reply:
Is this working on XML directly with ET, or in the RAVEN-defined heirarchical trees? Because we can use EDDI format trees (from WORKBENCH) as well, I would hesitate to strictly depend on the XML parsing.

Reply:
@PaulTalbot-INL This uses ET. I'm pretty confident in the functionality of the parsing for now, but it's not the easiest to work with, and the syntax can be a bit picky and hard to read. So if the EDDI format trees are significantly easier, it might be worthwhile to switch over for the sake of future edits and new tests. What's the best place to get more info on those?

Reply:
In RAVEN utils we have the TreeStructure, which behaves like XML and mimics ET in terms of methods, but is designed to flexibly load in different heirarchical structures (XML, EDDI, HIT, etc). It may not be too hard to load the XML into the tree structure and use that instead of parsing the XML directly.

See https://github.com/idaholab/raven/blob/devel/ravenframework/utils/TreeStructure.py.

Reply:
just to clarify, is the suggestion here to modify the `create_componentsets_in_HERON( )` method to output a `TreeStructure` object rather than an `ElementTree` object? I think that could be valuable, though I feel like might be out of scope for this particular PR. might be worth starting a different one to change that functionality.

also wanted to note that the input to the `create_componentsets_in_HERON( )` method accepts a 'str' path to some sort of HERON input script, so it could theoretically accept either a `.xml` or potentially `.heron` (for Workbench) file which it can modify. to actually _write_ to the new file, this would involve a new integration test or modifying the existing ones...

Reply:
I see your point. I'm fine with using XML and ET for now. We've been wanting to do something to allow InputSpec on-the-fly instantiation for dynamic component building, but we haven't figured out how to make the ends meet on that yet. This is fine with ET.

Reply:
@caleb-sitton-inl seems like it's best to keep as is, but also it might be a good idea to start an issue for Feature Request regarding switching to dynamic `TreeStructure` elements rather than ET